src/command.zig

```diff
@@ -16,32 +18,100 @@ pub const Command = enum {
     Unknown,
 };
 
-pub fn handle_commands(cmd: Command, params: ?[]const u8) !void {
-    switch (cmd) {
-        .List => {
-            try handle_list();
-        },
-        .Install => {
-            try install_version(params);
-        },
-        .Use => {
-            try use_version(params);
-        },
-        .Default => {
-            try set_default();
-        },
-        .Version => {
-            try get_version();
-        },
-        .Help => {
-            try display_help();
-        },
-        .Unknown => {
-            try handle_unknown();
-        },
+const CommandData = struct {
+    cmd: Command = .Unknown,
+    param: ?[]const u8 = null,
+};
+
+const CommandOption = struct {
+    short_handle: ?[]const u8,
+    handle: []const u8,
+    cmd: Command,
+};
+
+/// now all available commands
+const command_opts = [_]CommandOption{
+    .{ .short_handle = "ls", .handle = "list", .cmd = Command.List },
+    .{ .short_handle = "i", .handle = "install", .cmd = Command.Install },
+    .{ .short_handle = null, .handle = "use", .cmd = Command.Use },
+    .{ .short_handle = null, .handle = "--version", .cmd = Command.Version },
+    .{ .short_handle = null, .handle = "--help", .cmd = Command.Help },
+    .{ .short_handle = null, .handle = "--default", .cmd = Command.Default },
+};
+
+/// parse command and handle commands
+pub fn handleCommand(params: []const []const u8) !void {
```

Comment:
What do you think of write it like this?

```zig
pub fn handle_command(params: []const []const u8) !void {
    if (builtin.os.tag != .windows) {
        const command = std.fs.path.basename(params[0]);
        if (std.mem.eql(u8, command, "zig")) {
            try handle_alias(params);
        }
    }

    const commandData = get_command_data(params[1..]) catch |err| {
        // handle error appropriately
        return err;
    };

    switch (commandData.cmd) {
        .List => try handle_list(),
        .Install => try install_version(commandData.param),
        .Use => try use_version(commandData.param),
        .Default => try set_default(),
        .Version => try get_version(),
        .Help => try display_help(),
        .Unknown => try handle_unknown(),
    }
}

fn get_command_data(args: []const []const u8) !CommandData {
    for (args) |arg, index| {
        const opt = findOption(arg) orelse continue;

        return CommandData{
            .cmd = opt.cmd,
            .param = if (index + 1 < args.len) args[index + 1] else null,
        };
    }
    return CommandData{ .cmd = .Unknown };
}

fn find_option(arg: []const u8) ?OptionType {
    for (command_opts) |opt| {
        if (std.mem.eql(u8, arg, opt.handle)) return opt;
        if (opt.short_handle) |short_handle| {
            if (std.mem.eql(u8, arg, short_handle)) return opt;
        }
    }
    return null;
}
```

Reply:
The logic of this code is not complicated, I don't think it needs to be disassembled

Reply:
This is also the reason why I merged the original functions into one